### PR TITLE
Add recursive get data call

### PR DESCRIPTION
**Pull request by cfconrad**
when requesting `curl autoinst_url . "/data"` and the requested URI
points to an directory, we do now retrieve the directory recursive.
The returned data is a cpio-achive.

In action: http://cfconrad-vm.qa.suse.de/tests/4620/file/serial_terminal.txt

### CODE REVIEW

File: t/07-commands.t [hunk 1]
```diff
@@ -119,6 +123,31 @@ subtest 'web socket route' => sub {
...
 
+subtest 'data api' => sub {
+    my $job = 'Hallo';
+    $t->get_ok("$base_url/$job/data");
+    $t->status_is(200);
+    $t->content_type_is("application/x-cpio");
+
+    $t->get_ok("$base_url/$job/data/mod1");
+    $t->status_is(200);
+    $t->content_type_is("application/x-cpio");
```
Comment (cfconrad) [inline t/07-commands.t]: I would like to add a check, if the achieve really contains all files.
How do I check make sure, that cpio is available?

Comment (okurz) [reply t/07-commands.t]: do you mean if the cpio command line tool is available in the environment that runs the tests? As "cpio" is included by default even in the docker containers "opensuse/tumbleweed" or even "mini/base" (alpine) I would not be concerned about environments missing it but still check that the command exists

Comment (okurz) [reply t/07-commands.t]: Probably better:

```
use File::Which;
ok(-x which 'cpio');
```

Comment (cfconrad) [reply t/07-commands.t]: exact that was my question. Thanks a lot!

**Pull request merged**